### PR TITLE
[Fix #7338] Fix finding node as the base for indentation

### LINKED ISSUE #7338: False positive for `MultilineMethodCallIndentation` with `[]` on the left side of assignment
There is false positive for `MultilineMethodCallIndentation` with `[]` on the left side of assignment.

--------

## Expected behavior

Without offenses.

## Actual behavior

With offenses.

## Steps to reproduce the problem

```yaml
# .rubocop.yml
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

```ruby
# test.rb
# frozen_string_literal: true

params[:key] =
  foo
    .bar
    .baz
```

```
> bundle exec rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:5:5: C: Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression in an assignment spanning multiple lines.
    .bar
    ^^^^
test.rb:6:5: C: Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression in an assignment spanning multiple lines.
    .baz
    ^^^^
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
0.74.0 (using Parser 2.6.4.0, running on ruby 2.6.4 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by Neodelf**
…gnment

**Replace this text with a summary of the changes in your PR.
The more detailed you are, the better.**

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ### Bug fixes
 
+* [#7338](https://github.com/rubocop-hq/rubocop/issues/7338): Allow using indentation for variable's methods after assignment. ([@neodelf][])
```
Comment (bbatsov) [inline CHANGELOG.md]: Looking at the issue report this description seems a bit misleading - it seem the bug exists only if you have `[]` on the lhs.

File: lib/rubocop/cop/layout/multiline_method_call_indentation.rb [hunk 2]
```diff
@@ -214,6 +214,18 @@ def operator_rhs?(node, receiver)
...
             within_node?(receiver, node.first_argument)
         end
+
+        def relative_indentation(lhs, node)
+          indentation = indentation(lhs) + correct_indentation(node)
+          indentation += indentation(node) if node_indentationed?(lhs, node)
+          indentation
+        end
+
+        def node_indentationed?(lhs, node)
```
Comment (bbatsov) [inline lib/rubocop/cop/layout/multiline_method_call_indentation.rb]: Guess you meant to write `indented?`, but I really can't tell what this means in the context of a node. :-)

Comment (Neodelf) [reply lib/rubocop/cop/layout/multiline_method_call_indentation.rb]: Renaming at the night was a bad idea)
I meant that the node already has the indentation. Is `node_already_indented` would be better?

File: CHANGELOG.md (hunk 1 above)
Comment (Neodelf) [reply CHANGELOG.md]: Is `Allow using indentation for variable's methods if you have [] on the LHS.` would be better?

File: lib/rubocop/cop/layout/multiline_method_call_indentation.rb [hunk 3]
```diff
@@ -214,6 +214,18 @@ def operator_rhs?(node, receiver)
           node.operator_method? && node.arguments? &&
             within_node?(receiver, node.first_argument)
         end
+
+        def relative_indentation(lhs, node)
+          indentation = indentation(lhs) + correct_indentation(node)
+          indentation += indentation(node) if node_indentationed?(lhs, node)
```
Comment (buehmann) [inline lib/rubocop/cop/layout/multiline_method_call_indentation.rb]: Adding two indentations can't be correct in general.

Here is an example where it does not work:
```ruby
module Foo
  class Bar
    def foo
      params[:key] =
        foo
        .bar
        .baz
    end
  end
end
```
autocorrects to
```
7352.rb:6:9: C: [Corrected] Layout/MultilineMethodCallIndentation: Use 2 (not 2) spaces for indenting an expression in an assignment spanning multiple lines.
        .bar
        ^^^^
7352.rb:7:9: C: [Corrected] Layout/MultilineMethodCallIndentation: Use 2 (not 2) spaces for indenting an expression in an assignment spanning multiple lines.
        .baz
        ^^^^

1 file inspected, 5 offenses detected, 4 offenses corrected
====================
# frozen_string_literal: true

module Foo
  class Bar
    def foo
      params[:key] =
        foo
                .bar
                .baz
    end
  end
end
```
```

File: CHANGELOG.md (hunk 1 above)
Comment (Neodelf) [reply CHANGELOG.md]: Outdated

File: lib/rubocop/cop/layout/multiline_method_call_indentation.rb (hunk 2 above)
Comment (Neodelf) [reply lib/rubocop/cop/layout/multiline_method_call_indentation.rb]: Outdated

File: lib/rubocop/cop/layout/multiline_method_call_indentation.rb (hunk 3 above)
Comment (Neodelf) [reply lib/rubocop/cop/layout/multiline_method_call_indentation.rb]: Thank you for your feedback. It helped me to rethink the issue.

Comment (Neodelf) [reply lib/rubocop/cop/layout/multiline_method_call_indentation.rb]: Outdated

**Comment by Neodelf** [comment]
I've updated the PR. Could you please check it?

**Comment by Neodelf** [comment]
@bbatsov @buehmann Guys, what do you think about the PR?

**Comment by AlexWayfer** [comment]
By the way, there are conflicts already.

**Comment by Neodelf** [comment]
@AlexWayfer @bbatsov @buehmann 
Hi there! The PR is very old. So I have updated it. If the issue is still actual would we continue work with it?

**Comment by AlexWayfer** [comment]
> Hi there! The PR is very old. So I have updated it.

Thank you.



> If the issue is still actual would we continue work with it?

The issue is still open, so I guess yes, it's actual. The best way to be sure is adding auto-tests for your changes.

**Comment by Neodelf** [comment]
@AlexWayfer Thank you for your response!
I missed it due to rebasing. I will fix it a bit later

**Comment by kbarrette** [comment]
@Neodelf @AlexWayfer any motion on this? I'd love to see this fixed!

**Comment by bbatsov** [comment]
I'm closing this PR due to no recent activity. Feel free to re-open it if you ever come back to it.

**Pull request closed without merging**